HiP-DataStore/Controllers/ExhibitsController.cs

```diff
@@ -193,6 +195,73 @@ public IActionResult GetById(int id, DateTimeOffset? timestamp = null)
             return NoContent();
         }
 
+        [HttpGet("rating/{id}")]
+        [ProducesResponseType(typeof(RatingResult), 200)]
+        [ProducesResponseType(400)]
+        [ProducesResponseType(404)]
+        public IActionResult GetRating(int id)
+        {
+            if (!ModelState.IsValid)
+                return BadRequest(ModelState);
+
+            var exhibit = _db.Database.GetCollection<Exhibit>(ResourceType.Exhibit.Name)
+                                      .AsQueryable()
+                                      .FirstOrDefault(x => x.Id == id);
+
+            if (exhibit == null || _entityIndex.Status(ResourceType.Exhibit, id) != ContentStatus.Published)
+                return NotFound(ErrorMessages.ExhibitNotFoundOrNotPublished(id));
+
+            var result = new RatingResult()
+            {
+                Id = id,
+                Average = _ratingIndex.Average(ResourceType.Exhibit, id),
+                Count = _ratingIndex.Count(ResourceType.Exhibit, id)
+            };
+
+            return Ok(result);
+        }
+
+        [HttpPost("rating/{id}")]
+        [ProducesResponseType(typeof(int),201)]
+        [ProducesResponseType(400)]
+        [ProducesResponseType(404)]
+        public async Task<IActionResult> PostRating(int id,RatingArgs args)
+        {
+            if (!ModelState.IsValid)
+                return BadRequest(ModelState);
+
+            var exhibit = _db.Database.GetCollection<Exhibit>(ResourceType.Exhibit.Name)
```

Comment:
Remember we shouldn't rely on the MongoDB database when handling PUT- or POST-requests as it may not be consistent with the Event Store. Normally, things go like this:
- user posts a rating of 3 (old rating = null) => RatingAdded-event <span>#</span>1 is emitted
- Mongo cache DB receives RatingAdded-event <span>#</span>1
- user posts a rating of 4 (old rating = 3) => RatingAdded-event <span>#</span>2 is emitted
- Mongo cache DB receives RatingAdded-event <span>#</span>2

But the order is not predictable, so something like this may also happen:
- user posts a rating of 3 (old rating = null) => RatingAdded-event <span>#</span>1 is emitted
- user posts a rating of 4 (**old rating = null**) => RatingAdded-event <span>#</span>2 is emitted
- Mongo cache DB receives RatingAdded-event <span>#</span>1
- Mongo cache DB receives RatingAdded-event <span>#</span>2

...and this may mess up your rating calculations.